src/R/Components/Impl/Controller/IMenuCommand.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Components.Controller {
+    /// <summary>
+    /// An object that implements a single command. 
+    /// </summary>
+    public interface IMenuCommand {
```

Comment:
NeedCheckout is generally obsolete since VS no longer requires manual SCC interactions. There is a task to reshuffle dependencies. Perhaps we could still make existing interface work? The difference is in missing ```Guid group, int id``` which in ICommand allow command implementation to register for multiple IDs. However, the command doesn't have to check for them as controller does the work.